test/unit/hpa_background_thread.c

```diff
@@ -128,6 +128,7 @@ TEST_BEGIN(test_hpa_background_thread_purges) {
 	test_skip_if(!config_stats);
 	test_skip_if(!hpa_supported());
 	test_skip_if(!have_background_thread);
+	test_skip_if(san_enabled());
```

Comment:
Could you please explain why we skip here?

Reply:
Yep will add comments.

This test focuses on the hpa purging behavior, while the guarded pages cannot go through hpa, making the test irrelevant.